### PR TITLE
[Feature] Add support for PG Query Log exporter

### PR DESCRIPTION
**Pull request by bhupendray-yb**
### Description:
Added support for DB Query logs exporter(https://docs.yugabyte.com/preview/yugabyte-cloud/cloud-monitor/logging-export/)

### Changelog:
- Added support for 4 subcommands for db query logging:
    - Enable logging
    - Disable logging
    - Describe config
    - Update logging config
- Added unit test for all commands

#### Enabling query log exporter
```shell
~/Developer/ybm-cli on feat/db_query_log_exporter                                                                                                       took 4s at 12:21:08
> make build && ./ybm cluster db-query-logging enable \
--cluster-name "bhupendray-cluster" \
--integration-name bhupendray-datadog-logging \
--log-line-prefix "%m :%r :%u @ %d :[%p] : %a :" \
--log-min-duration-statement 30 \
--log-connections true \
--log-duration false \
--log-error-verbosity DEFAULT \
--log-statement MOD
go build -ldflags="-X 'main.version=v0.1.0'" -o ybm
A newer version is available. Please upgrade to the latest version v0.1.20
State      Integration ID                            Log Config
ENABLING   7eb1b884-600e-4a91-a92c-9ee79b754e11   {"debug_print_plan":false,"log_connections":true,"log_disconnections":false,"log_duration":false,"log_error_verbosity":"DEFAULT","log_line_prefix":"%m :%r :%u @ %d :[%p] : %a :","log_min_duration_statement":30,"log_min_error_statement":"ERROR","log_statement":"MOD"}

```

#### Describe log exporter config:
```shell
~/Developer/ybm-cli on feat/db_query_log_exporter                                                                                                       took 7s at 12:09:00
> make build && ./ybm cluster db-query-logging describe --cluster-name "bhupendray-cluster"
go build -ldflags="-X 'main.version=v0.1.0'" -o ybm
A newer version is available. Please upgrade to the latest version v0.1.20
State       Integration ID                            Log Config
DISABLING   7eb1b884-600e-4a91-a92c-9ee79b754e11   {"debug_print_plan":false,"log_connections":true,"log_disconnections":false,"log_duration":false,"log_error_verbosity":"DEFAULT","log_line_prefix":"%m :%r :%u @ %d :[%p] : %a :","log_min_duration_statement":30,"log_min_error_statement":"ERROR","log_statement":"MOD"}

```

#### Disable query logs:
```bash
~/Developer/ybm-cli on feat/db_query_log_exporter                                                                                                           INT at 12:08:37
> make build && ./ybm cluster db-query-logging disable \
--cluster-name "bhupendray-cluster"
go build -ldflags="-X 'main.version=v0.1.0'" -o ybm
A newer version is available. Please upgrade to the latest version v0.1.20
Disabling DB query log config for the cluster, this may take a few minutes...
You can check the status via $ ybm cluster db-query-log-exporter describe --cluster-name bhupendray-cluster
```

#### Edit log exporter config:
```shell
~/Developer/ybm-cli on feat/db_query_log_exporter                                                                                                       took 5s at 11:19:37
> make build && ./ybm cluster db-query-logging update-config \
--cluster-name "bhupendray-cluster" \
--integration-name bhupendray-datadog-logging \
--log-line-prefix "%m :%r :%u @ %d :[%p] :" \
--log-min-duration-statement 60 \
--log-connections false \
--log-duration true \
--log-error-verbosity DEFAULT \
--log-statement ALL
go build -ldflags="-X 'main.version=v0.1.0'" -o ybm
A newer version is available. Please upgrade to the latest version v0.1.20
State     Integration ID                            Log Config
ACTIVE    7eb1b884-600e-4a91-a92c-9ee79b754e11  {"debug_print_plan":false,"log_connections":false,"log_disconnections":false,"log_duration":true,"log_error_verbosity":"DEFAULT","log_line_prefix":"%m :%r :%u @ %d :[%p] :","log_min_duration_statement":60,"log_min_error_statement":"ERROR","log_statement":"ALL"}
```

### CODE REVIEW

File: cmd/cluster/log-exporter/query_log_exporter.go
Comment (atharva-yb) [inline cmd/cluster/log-exporter/query_log_exporter.go:34]: Please keep the command consistent with DB Audit logging.
DB Audit log has command : `db-audit-logs-exporter`.

CC: @Siddarth-Baldwa @bansal01yash

Comment (Siddarth-Baldwa) [reply cmd/cluster/log-exporter/query_log_exporter.go:34]: @atharva-yb talked to the PMs about this. We need to keep `db-audit-export` commands under the cluster command as is being done for pg logs (in this PR)

Comment (atharva-yb) [reply cmd/cluster/log-exporter/query_log_exporter.go:34]: True. I was referring the exact word of command.
If we are going to use `ybm cluster db-audit-logs-exporter` for DB audit, we should use `ybm cluster db-query-logs-exporter` for Query logs.
If we want to use `ybm cluster db-query-logging` for Query then we need to modify `ybm cluster db-audit-logging` for Audit.
Let's keep it consistent.

Comment (bansal01yash) [reply cmd/cluster/log-exporter/query_log_exporter.go:34]: @bhupendray-yb Can you make it `db-query-log-exporter`
@atharva-yb since we need to slightly refactor audit to be under cluster anyways, then we can rename audit to `db-audit-log-exporter` . Just to be a little consistent with API. 
Note: change from logs -> log

Comment (atharva-yb) [reply cmd/cluster/log-exporter/query_log_exporter.go:34]: Any command works. Just make them similar for Audit and Query. 👍

File: cmd/cluster/log-exporter/query_log_exporter.go
Comment (posriniv) [inline cmd/cluster/log-exporter/query_log_exporter.go:34]: Is this released to the public already? If so, we can't really change it. If it is not released, then we can go ahead. Please let me know.

Comment (bansal01yash) [reply cmd/cluster/log-exporter/query_log_exporter.go:34]: DB Query Log feature is GA, CLI for it is being developed as part of this PR.

File: cmd/cluster/log-exporter/query_log_exporter.go
Comment (bansal01yash) [inline cmd/cluster/log-exporter/query_log_exporter.go:37]: Nit: `DB Query Log exporter` -> `Database Query Logging` for all commands

**Review by bansal01yash — APPROVED** [review]
Minor nit. 
Apart from that please squash commits before merging once you get approval from owners.

File: internal/formatter/db_query_log_exporter.go
Comment (arishta-yb) [inline internal/formatter/db_query_log_exporter.go:70]: Can you add null check after fetching the values from any spec.

File: internal/client/client.go
Comment (arishta-yb) [inline internal/client/client.go:1660]: Please wrap the error with additional context, something liked: failed to get integration by name..?

**Review by arishta-yb — APPROVED** [review]
Left minor comments, mostly around readability and adding safety checks. Also, squash all the commits before merging to keep the commit history clean :)

File: internal/formatter/db_query_log_exporter.go
Comment (bhupendray-yb) [reply internal/formatter/db_query_log_exporter.go:77]: This change is no longer required in a subsequent part of this PR.
Here: https://github.com/yugabyte/ybm-cli/pull/257/files#diff-2851f27c199dad02e5ea784727fcea939caba1a4b600ebf6f5c2258bc7f7d17bL76

So is it fine if I don't make this change, as it's anyways going to be removed?
But I got the intent of the comment, will follow it in future code changes.

**Comment by bhupendray-yb** [comment]
I will resolve the comments.
> Squash the commit

@bansal01yash @arishta-yb do I need to do it manually? In merge PR option I see, it's automatically going to squash and then merge.

File: internal/client/client.go
Comment (bhupendray-yb) [reply internal/client/client.go:1660]: done.

File: internal/formatter/db_query_log_exporter.go
Comment (bhupendray-yb) [reply internal/formatter/db_query_log_exporter.go:70]: Same as above comment(i.e. it will be removed in subsequent PR).

Just for my understanding: is there a case when spec can be null? As per [API contract for DB Query logs](https://api-docs.yugabyte.com/docs/managed-apis/badf5234abfd9-create-a-database-query-logging-configuration-and-associate-with-cluster) all fields are required, so I did not added a check for it. Curious to know when can nil issue arise here?

File: cmd/cluster/log-exporter/query_log_exporter.go
Comment (bhupendray-yb) [reply cmd/cluster/log-exporter/query_log_exporter.go]: done

**Pull request merged**
